homeassistant/components/samsungtv/__init__.py

```diff
@@ -1 +1,60 @@
 """The Samsung TV integration."""
+from wakeonlan import BROADCAST_IP
+import voluptuous as vol
+
+from homeassistant.const import (
+    CONF_BROADCAST_ADDRESS,
+    CONF_HOST,
+    CONF_MAC,
+    CONF_NAME,
+    CONF_PORT,
+    CONF_TIMEOUT,
+)
+import homeassistant.helpers.config_validation as cv
+
+from .const import DEFAULT_NAME, DEFAULT_TIMEOUT, DOMAIN
+
+
+CONFIG_SCHEMA = vol.Schema(
+    {
+        DOMAIN: vol.All(
+            [
+                vol.Schema(
+                    {
+                        vol.Required(CONF_HOST): cv.string,
+                        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+                        vol.Optional(CONF_PORT): cv.port,
+                        vol.Optional(CONF_MAC): cv.string,
+                        vol.Optional(
+                            CONF_TIMEOUT, default=DEFAULT_TIMEOUT
+                        ): cv.positive_int,
+                        vol.Optional(
+                            CONF_BROADCAST_ADDRESS, default=BROADCAST_IP
+                        ): cv.string,
+                    }
+                )
+            ]
+        )
+    },
+    extra=vol.ALLOW_EXTRA,
+)
+
+
+async def async_setup(hass, config):
+    """Set up the Samsung TV integration."""
+    if DOMAIN in config:
+        for entry_config in config[DOMAIN]:
+            await hass.config_entries.flow.async_init(
+                DOMAIN, context={"source": "user"}, data=entry_config
```

Comment:
This should be source import.

Reply:
It was some time ago but it was you who requested to use the user step :-)
 https://github.com/home-assistant/home-assistant/pull/28306#discussion_r343500937

Reply:
We still need an import step as a middle layer. From there pass the config to the user step.